### PR TITLE
Add phpstan composer script

### PR DESCRIPTION
**Pull request by tswfi**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add PHPStan composer script.
| Type?             | improvement
| Category?         | PM
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | run `docker compose exec prestashop-git composer phpstan`
| UI Tests          | -
| Fixed issue or discussion?     | -
| Related PRs       | -
| Sponsor company   | -

Just for running phpstan easily while developing

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -285,6 +285,9 @@
         ],
         "unit-tests": [
             "@php -d date.timezone=UTC ./vendor/phpunit/phpunit/phpunit -c tests/Unit/phpunit.xml"
+        ],
+        "phpstan": [
```
Comment (kpodemski) [inline composer.json]: cs fixer wants to move this before "restore-test-db" 😓

**Review by nicosomb — COMMENTED** [review]
there is already a `makefile` recipe (`make phpstan`), which runs `./vendor/bin/phpstan analyse -c phpstan.neon.dist`

So, I think that the composer script should be the same, to keep consistancy.

File: composer.json (hunk 1 above)
Comment (tswfi) [reply composer.json]: Yup, closed laptop already. I'll fix it later :)

**Comment by tswfi** [comment]
> there is already a `makefile` recipe (`make phpstan`), which runs `./vendor/bin/phpstan analyse -c phpstan.neon.dist`

Oh, didn't even check Makefile.. I think this raises a question if the makefile should be removed completely and everything moved to composer scripts?

**Comment by tswfi** [comment]
These also need at least some documentation in devdocs for future reference (I think `test-all` was documented somewhere already). 
I can make a PR for the documentation after this gets merged.

And few scripts require descriptions: 

```
$ docker compose exec prestashop-git composer run-script
Script to run: 
  [composer-normalize         ] Normalize composer.json file, CI will check this
  [create-release             ] Create a release of PrestaShop, run the command with -h/--help argument for more information.
  [create-test-db             ] Create a Database for testing purposes
  [create-test-table-dumps    ] Create a dump of each table in the Database for testing purposes
  [git-hook-install           ] Runs the git-hook-install script as defined in composer.json
  [git-hook-uninstall         ] Runs the git-hook-uninstall script as defined in composer.json
  [integration-behaviour-tests] Runs the integration-behaviour-tests script as defined in composer.json
  [integration-tests          ] Runs the integration-tests script as defined in composer.json
  [phpstan                    ] Run PHPStan for the project
  [restore-test-db            ] Restore the whole DB in a single dump
  [restore-test-tables        ] Restore all tables one by one, only when they have been modified
  [test-all                   ] Launch all PHPUnit test suites
  [unit-tests                 ] Runs the unit-tests script as defined in composer.json
 > 
```

File: composer.json [hunk 2]
```diff
@@ -272,6 +275,10 @@
             "@php -d date.timezone=UTC -d memory_limit=-1 ./vendor/phpunit/phpunit/phpunit -c tests/Integration/phpunit.xml",
             "@php -d date.timezone=UTC ./vendor/phpunit/phpunit/phpunit -c tests/Integration/phpunit-isolated.xml"
         ],
+        "phpstan": [
+            "Composer\\Config::disableProcessTimeout",
+            "make phpstan"
```
Comment (nicosomb) [inline composer.json:280]: I think that composer scripts shouldn't depend on `make`. The previous version was ok for me (`./vendor/bin/phpstan analyse --memory-limit 1G -c phpstan.neon.dist`). 

`make` is not available on each platform. If I use composer scripts, I don't have to know if I have `make` or something else. 

I don't know if I'm clear 😉

Comment (tswfi) [reply composer.json:280]: Yeah, its a bit unclear now that there are two different places for these. 

I will raise this on the slack channel and try to get somekind of "vote" if Makefile should be deprecated and all scripts moved into composer scripts.

**Comment by tswfi** [comment]
Also changed this PR to draft until there is a consensus on how to proceed with this.

**Comment by tswfi** [comment]
this is too old. closing.

**Pull request closed without merging**